src/datatrove/pipeline/filters/gopher_quality_filter.py

```diff
@@ -66,10 +67,10 @@ def filter(self, doc: Document) -> bool | tuple[bool, str]:
         Returns: False if sample.text does not pass any of the the heuristic tests
 
         """
-        from nltk.tokenize import word_tokenize
-
         text = doc.text
-        words = word_tokenize(text)  # TODO we should use language id filter
+        language = doc.metadata["language"]
+        tokenizer = get_word_tokenizer(language)
+        words = tokenizer.tokenize(text)
```

Comment:
This is the proposed change that would be applied to all places where tokenization is performed. How do we want to change the API to provide backwards compatibility and handle cases where language is not provided in the document metadata?